src/k8s/pkg/k8sd/api/certificates_refresh.go

```diff
@@ -275,7 +256,23 @@ func isCertificateSigningRequestApproved(csr *certificatesv1.CertificateSigningR
 	return false, nil
 }
 
-// isCertificateSigningRequestIssued checks if the certificate signing request is issued.
-func isCertificateSigningRequestIssued(csr *certificatesv1.CertificateSigningRequest) bool {
-	return len(csr.Status.Certificate) > 0
+// verifyCSRAndSetPKI verifies the certificate signing request and sets the
+// certificate and key if the CSR is approved.
+func verifyCSRAndSetPKI(csr *certificatesv1.CertificateSigningRequest, keyPEM string, certificate, key *string) (bool, error) {
+	approved, err := isCertificateSigningRequestApprovedAndIssued(csr)
+	if err != nil {
+		return false, fmt.Errorf("%w: failed to validate csr: %w", errInvalidCSR, err)
```

Comment:
avoid wrapping two errors, also see the other comment. return a bool that specifies whether we are treating this as a retriable error or not.